src/core/io/writer/h5md_core.cpp

```diff
@@ -47,562 +52,340 @@ static void backup_file(const std::string &from, const std::string &to) {
-
-  id[0][particle_index][0] = current_particle.p.identity;
-  if (write_species)
-    typ[0][particle_index][0] = current_particle.p.type;
-  if (write_mass)
-    mass[0][particle_index][0] = current_particle.p.mass;
-  /* store folded particle positions. */
-  if (write_pos) {
-    Utils::Vector3d p = current_particle.r.p;
-    Utils::Vector3i i = current_particle.l.i;
-    fold_position(p, i, box_geo);
-
-    pos[0][particle_index][0] = p[0];
-    pos[0][particle_index][1] = p[1];
-    pos[0][particle_index][2] = p[2];
-    image[0][particle_index][0] = i[0];
-    image[0][particle_index][1] = i[1];
-    image[0][particle_index][2] = i[2];
-  }
-
-  if (write_vel) {
-    vel[0][particle_index][0] = current_particle.m.v[0];
-    vel[0][particle_index][1] = current_particle.m.v[1];
-    vel[0][particle_index][2] = current_particle.m.v[2];
+void hard_link(h5xx::file const &file, std::string from, std::string to) {
+  if (H5Lcreate_hard(file.hid(), from.c_str(), file.hid(), to.c_str(),
+                     H5P_DEFAULT, H5P_DEFAULT) < 0) {
+    throw std::runtime_error("Error creating hard link for " + to);
   }
-  if (write_force) {
-    f[0][particle_index][0] = current_particle.f.f[0];
-    f[0][particle_index][1] = current_particle.f.f[1];
-    f[0][particle_index][2] = current_particle.f.f[2];
-  }
-  if (write_charge) {
-#ifdef ELECTROSTATICS
-    charge[0][particle_index][0] = current_particle.p.q;
-#endif
-  }
-
-  if (!m_already_wrote_bonds) {
-    int nbonds_local = bond.shape()[1];
+}
 
-    for (auto const &b : current_particle.bonds()) {
-      auto const partner_ids = b.partner_ids();
-      if (partner_ids.size() == 1) {
-        bond.resize(boost::extents[1][nbonds_local + 1][2]);
-        bond[0][nbonds_local][0] = current_particle.p.identity;
-        bond[0][nbonds_local][1] = partner_ids[0];
-        nbonds_local++;
-      }
+void File::create_hard_links() {
+  std::string path_step = "particles/atoms/id/step";
+  std::string path_time = "particles/atoms/id/time";
+  for (auto &ds : H5MD_Specification::DATASETS) {
+    if (ds.name == "step" and ds.is_link) {
+      hard_link(m_h5md_file, path_step, ds.path());
+    } else if (ds.name == "time" and ds.is_link) {
+      hard_link(m_h5md_file, path_time, ds.path());
     }
   }
 }
 
-void File::Write(int write_dat, PartCfg &partCfg,
-                 const ParticleRange &particles) {
-  int num_particles_to_be_written = 0;
-  int n_part = 0;
-  if (m_write_ordered && this_node == 0) {
-    num_particles_to_be_written = partCfg.size();
-    n_part = partCfg.size();
-  } else if (m_write_ordered && this_node != 0)
-    return;
-  else if (!m_write_ordered) {
-    num_particles_to_be_written = particles.size();
-    n_part = boost::mpi::all_reduce(m_hdf5_comm, num_particles_to_be_written,
-                                    std::plus<int>());
-  }
+void File::create_new_file(const std::string &filename, BoxGeometry &geometry) {
+  if (this_node == 0)
```

Comment:
The `this_node` global can be dropped (`m_comm.rank()`), please do also remove the corresponding header to check that you catch all places.